xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-image/attachmentService.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+define('attachmentService', [
+  'jquery',
+  'xwiki-attachment-picker-solr-search',
+], function ($, SolrSearch) {
+
+  // Attachment cache
+  var attachmentsByQuery = {};
+
+  var clearCache = function () {
+    attachmentsByQuery = {
+      global: {},
+      local: {}
+    };
+  };
+
+  clearCache();
+
+  var getAttachments = function (query, isGlobal, force, editor, sort) {
+
+    return new Promise(function (resolve) {
+
+      var cached = attachmentsByQuery[(isGlobal ? "global" : "local")][query];
+
+      if (cached && !force) {
+        return resolve(cached);
+      }
+
+      const solrSearch = new SolrSearch({
+          limit: 100,
```

Comment:
Do we really need to fetch that many attachments?
We are calling `getAttachments` four times on each (debounced) keystroke, meaning a potential 400 entries (some of the as duplicates).
WDYT of lowering this to say, 40? I doubt anybody would go that low in the list.
Future improvement could be to lazily show additional results when the user reaches the end of the list.
And also to show a message indicating if this is the end of the list, or if there is additional results.